design/variadic_config_alt.md

```diff
@@ -0,0 +1,637 @@
+
+The purpose of adding a `repeatable` attribute to Config Items is to add the capability *EXTEND* resources.
+
+The existing Config Item concept will be augmented with a new property `repeatable` to indicated the value will be an array of values rather than a scalar. The value types will still inherit from the `type` field.
+
+To use these array values, a new method `ConfigOptionMap` will be added to the Replicated [Config Context](https://kots.io/reference/template-functions/config-context/) template functions to provide a `pipeline` output that can be used in conjunction with `range` in a Golang Text Tempalate to iterate over values.
+
+### `reapeatable` Config Groups
+
+The purpose of adding a `repeatable` attribute to Config Groups is to add the ability to *COPY* collections of resources/config.
+
+The existing Config Group concept will be augmented with a new property `repeatable` to indicated the values in each Config Item will be an array. 
+In the case where this item also has the repeatable attribute, values will be flatted into a array with a stride defined by the number of group instances.
+Value names will also reflect both the index in the group and the index in the item.
+
+To use these array values, a new method `ConfigGroupList` will be added to the Replicated [Config Context](https://kots.io/reference/template-functions/config-context/) template functions to provide a `pipeline` output that can be used in conjunction with `range` in a Golang Text Tempalate to iterate over values.
+
+## Detailed Design
+
+While the design considered here is presented in an interleaved fashion, this proposal suggests that work be broken up in the following tasking:
+1. Repeatable Config Items
+1. Repeatable Config Groups
+
+The first consideration is how the revised API will look to Vendors using these features in their application.
+
+### Example Revised Kotskind Resources 
+
+Vendors will use the revised Config Spec to define templateGroups and repeatable Config Items. 
+Values are inserted by the Kots and returned as part of the API for creating the ConfigValues spec.
+Below is a representative resource.
+
+#### Config
+```yaml
+apiVersion: kots.io/v1beta1 
+kind: Config 
+metadata: 
+  creationTimestamp: null 
+  name: config-sample 
+spec: 
+  # UNCHANGED
+  groups: 
+  - name: nginx_settings 
+    title: Nginx Configs 
+    description: Config to serve as an example for creating your own
+    items: 
+    - name: "nginx_port"
+      type: "text"
+      title: "Nginx port"
+      default: "80"
+      value: ""
+
+    # NEW!
+    # This is a repeatable Config Item
+    - name: "static_files"
+      type: file
+      title: "Static Assets"
+      repeatable: true      # NEW! Tells the UI/Kots to expect an array
+      minimumCount: 3       # NEW! Not sure if this is needed here but including for discussion
+      repeatValues:         # NEW! Returned to the API filled in from the CLI/console
+      - "encoded file value one"
```

Comment:
I think that we will need to have some kind of "ID" as the other repeatValues in the example to have the possibility to add the filename from an end user perspective.

Reply:
This has been added in with the updated design